drivers/coreaudio/audio_driver_coreaudio.cpp

```diff
@@ -66,6 +66,11 @@ OSStatus AudioDriverCoreAudio::output_device_address_cb(AudioObjectID inObjectID
 
 	return noErr;
 }
+
+// Switch to kAudioObjectPropertyElementMain everywhere to remove deprecated warnings
+#if __MAC_OS_X_VERSION_MAX_ALLOWED < __MAC_12_0
+#define kAudioObjectPropertyElementMain kAudioObjectPropertyElementMaster
```

Comment:
How is this related to this? Should be in a separate PR if it's not specifically because of the metal changes

Reply:
It generates a number of warnings when compiling for arm64 with Metal enabled – happy to remove it, but it was a lot of noise.

Reply:
As long as it's from this specific change then it's fine :)

Reply:
My suggestion would be to split this into a separate PR and merge it. It seems like an unrelated change even if it is triggered by the Metal build.

Reply:
Yeah I agree a separate PR with that fix could be good, so we can also backport it to 4.3.x.
It's likely that this kind of deprecation warning will find its way even in non-Metal builds, I'm not sure why Metal specifically would enable it.